odoo/service/server.py

```diff
@@ -621,6 +621,11 @@ def stop(self):
 
         odoo.sql_db.close_all()
 
+        current_process = psutil.Process()
+        children = current_process.children(recursive=False)
+        for child in children:
+            _logger.info('A child process was found, pid is %s, process may hang', child)
+
```

Comment:
This part is maybe not needed but even for http worker it could help to understand what current worker is blocking the server shutdown. In this case it would have gave an obvious answer on why the server was stuck.

Reply:
I'm +1 on merging the log, that's a good step in the right direction